### PR TITLE
Don't rewrite fragment links

### PR DESCRIPTION
**Pull request by jonathansampson**
Resolves https://github.com/brave-experiments/SpeedReader/issues/13

## Submitter Checklist:

- [x] Submitted a [ticket](https://github.com/brave/brave-browser/issues) for my issue if one did not already exist.
- [x] Used Github [auto-closing keywords](https://help.github.com/articles/closing-issues-via-commit-messages/) in the commit message.
- [ ] Added/updated tests for this change (for new code or code which already has tests).
- Verified that these changes build without errors on
  - [ ] Android
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- Verified that these changes pass automated tests (unit, browser, security tests) on
  - [ ] iOS
  - [ ] Linux
  - [ ] macOS
  - [x] Windows
- [x] Verified that all lint errors/warnings are resolved (`npm run lint`)
- [x] Ran `git rebase master` (if needed).
- [x] Ran `git rebase -i` to squash commits (if needed).
- [x] Tagged reviewers and labelled the pull request as needed.
- [x] Request a security/privacy review as needed.
- [ ] Add appropriate QA labels (QA/Yes or QA/No) to include the closed issue in milestone
- [ ] Public documentation has been updated as necessary. For instance:
  - [ ] https://github.com/brave/brave-browser/wiki/Deviations-from-Chromium-(features-we-disable-or-remove)
  - [ ] https://github.com/brave/brave-browser/wiki/Proxy-redirected-URLs
  - [ ] https://github.com/brave/brave-browser/wiki/Fingerprinting-Protections
  - [ ] https://github.com/brave/brave-browser/wiki/Brave%E2%80%99s-Use-of-Referral-Codes
  - [ ] https://github.com/brave/brave-browser/wiki/Custom-Headers
  - [ ] https://github.com/brave/brave-browser/wiki/Web-compatibility-issues-with-tracking-protection
  - [ ] https://github.com/brave/brave-browser/wiki/QA-Guide

## Test Plan: 

1. With SpeedReader enabled, visit a supported page which uses `[href^=#foo]` anchors
2. Click one of these fragment links

The page should scroll to the appropriate element.

## Reviewer Checklist:

- [ ] New files have MPL-2.0 license header.
- [ ] Request a security/privacy review as needed.
- [ ] Adequate test coverage exists to prevent regressions 
- [ ] Verify test plan is specified in PR before merging to source

## After-merge Checklist:

- [ ] The associated issue milestone is set to the smallest version that the
  changes has landed on.
- [ ] All relevant documentation has been updated.

### CODE REVIEW

File: components/speedreader/rust/lib/src/rewriter_config_builder.rs [hunk 1]
```diff
@@ -227,6 +227,7 @@ fn correct_relative_links(
             if !href.starts_with("http://")
                 && !href.starts_with("https://")
                 && !href.starts_with("//")
+                && !href.starts_with("#")
```
Comment (iefremov) [inline components/speedreader/rust/lib/src/rewriter_config_builder.rs:230]: wondering if we can add a test for this

**Comment by AndriusA** [comment]
thanks for the fix - added some tests as suggested, but realised we don't actually need it since the URL doesn't change when rewriting in the current set up, so disabled that bit

**Comment by jonathansampson** [comment]
@AndriusA Can you clarify "the URL doesn't change…in the current setup"?

I'm seeing the following:

A link to `#fizz` on `foo.com/bar` redirects to `foo.com/#fizz` instead of `foo.com/bar#fizz`.

**Comment by AndriusA** [comment]
I meant that the main page URL does not change because we rewrite content, so we shouldn't need to modify the URLs to keep them working - relative, absolute and anchor links should work without touching them I think

**Comment by jonathansampson** [comment]
Ah, yes. That makes sense. As long as we don't drop the element ID via `preprocess`, the relative-fragment-links should _just work_ 🙂

**Comment by jonathansampson** [comment]
@AndriusA Safe to merge/close at this point?

**Pull request merged**